### DISCUSSION TITLE
Optional arguments for peripheral methods

### DISCUSSION THREAD

**Discussion by Mathilde411** [cc-tweaked/CC-Tweaked#837] [Q&A] [answered]
Hello again ! I want to create a method for a peripheral that has a nullable parameter.
I tried that:
```java
@LuaFunction
public final void write(ByteBuffer bytes, @Nullable String label) throws LuaException {
          byte[] data = Util.getByteBufferArray(bytes);
        if (data.length > Config.MAX_MAG_CARD_DATA)
            throw new LuaException("You can't put more than " + Config.MAX_MAG_CARD_DATA + " characters in a mag card.");
        getTile().writeCard(data, label);
    }

    @LuaFunction
    public final void write(ByteBuffer bytes) throws LuaException {
        this.write(bytes, null);
    }
```
But I always get `bad argument #2 (string excpected, got nil)` when using the method ingame
So, is there a way to have optional arguments to peripheral methods ?

**Comment by SquidDev** [comment, marked as answer]
`@Nullable` isn't available at runtime I don't believe, and so we can't check for it. `Optional<String>` should work.

**Reply by Mathilde411** [reply]
That's weird
IntelliJ warns me that Optional should be avoided as it has an unpredictable behavior.

Anyways, It's working like a charm !
Thanks ^^

**Reply by SquidDev** [reply]
Yeah, it's not _fantastic_, but I'm not convinced it's the anti-pattern IntelliJ warns it is. Either way, it's pretty much the only decent option available, so ¯\\\_(ツ)\_/¯.
